### PR TITLE
Allow metadata for launcher items

### LINKED ISSUE #7652: Pass Kernel Metadata to Launcher
## Description
We should allow kernel optional metadata in `ILauncher.IItem` and it to the launcher in the console and notebook extensions for implementations of the launcher that can utilize the metadata in rendering the launcher.

### PR DESCRIPTION
**Pull request by blink1073**
## References
Fixes #7652.

## Code changes
Allows optional metadata to be passed to launcher items, and passes in the kernel metadata for the items created by notebooks and consoles.

## User-facing changes
None, for the current implementation of the launcher.

## Backwards-incompatible changes
None

### CODE REVIEW

File: packages/console-extension/src/index.ts [hunk 1]
```diff
@@ -190,7 +191,8 @@ async function activateConsole(
               args: { isLauncher: true, kernelPreference: { name } },
               category: 'Console',
               rank,
-              kernelIconUrl
+              kernelIconUrl,
+              metadata: spec.metadata
```
Comment (jasongrout) [inline packages/console-extension/src/index.ts]: It feels like we should namespace the kernel spec metadata so items can have other metadata too. For example, `metadata: {kernel: spec.metadata}` or something.

Comment (afshin) [reply packages/console-extension/src/index.ts]: Hm. That's a good point. 👍

File: packages/launcher/src/index.tsx [hunk 2]
```diff
@@ -339,6 +339,12 @@ export namespace ILauncher {
      * spec.
      */
     kernelIconUrl?: string;
+
+    /**
+     * Metadata about the item.  This can be used by the launcher to
+     * affect how the item is displayed.
+     */
+    metadata?: ReadonlyJSONObject;
```
Comment (jasongrout) [inline packages/launcher/src/index.tsx]: Perhaps the default implementation should display this metadata somewhere? It feels a bit weird to include metadata without using it anywhere. Presumably you have an overridden launcher that actually is using the metadata?

Comment (blink1073) [reply packages/launcher/src/index.tsx]: Yes, this change was meant to allow experimentation without needing to override core extensions like `notebook` and `console`.

File: packages/console-extension/src/index.ts (hunk 1 above)
Comment (blink1073) [reply packages/console-extension/src/index.ts]: That's fair, I'll make the change.

File: packages/launcher/src/index.tsx (hunk 2 above)
Comment (jasongrout) [reply packages/launcher/src/index.tsx]: Sure, okay.

**Comment by jasongrout** [comment]
Looks like the failures are legit

**Pull request merged**